Frends.Community.SQL/Frends.Community.SQL.cs

```diff
@@ -226,6 +227,91 @@ private static Encoding GetEncoding(FileEncoding optionsFileEncoding, bool optio
+                    using (var sqlBulkCopy = new SqlBulkCopy(connection.ConnectionString, flagEnum))
+                    {
+                        sqlBulkCopy.BulkCopyTimeout = options.CommandTimeoutSeconds;
+                        sqlBulkCopy.DestinationTableName = input.TableName;
+
+                        await sqlBulkCopy.WriteToServerAsync(input.InputData, cancellationToken).ConfigureAwait(false);
+
+                        return sqlBulkCopy.RowsCopiedCount();
+                    }
+                }
+
+                using (var transaction =
+                    options.SqlTransactionIsolationLevel == SqlTransactionIsolationLevel.Default
+                        ? connection.BeginTransaction()
+                        : connection.BeginTransaction(options.SqlTransactionIsolationLevel.GetSqlTransactionIsolationLevel()))
+                {
+
+
+                    int rowsCopyCount;
+                    using (var sqlBulkCopy = new SqlBulkCopy(connection, flagEnum, transaction))
+                    {
+                        sqlBulkCopy.BulkCopyTimeout = options.CommandTimeoutSeconds;
+                        sqlBulkCopy.DestinationTableName = input.TableName;
+
+                        await sqlBulkCopy.WriteToServerAsync(input.InputData, cancellationToken).ConfigureAwait(false);
+
+                        rowsCopyCount = sqlBulkCopy.RowsCopiedCount();
+                    }
+                    transaction.Commit();
+                    return rowsCopyCount;
+                }
+            }
+        }
+
+        // Cast Enum value to another Enum type by value name
+        // Example: Frends.Community.SQL.SqlTransactionIsolationLevel -> System.Data.IsolationLevel
+        // From https://github.com/FrendsPlatform/Frends.Sql/blob/master/Frends.Sql/Extensions.cs
+        private static T GetEnum<T>(Enum enumValue)
+        {
+            return (T)Enum.Parse(typeof(T), enumValue.ToString());
+        }
+        // Cast Frends.Community.SQL.SqlTransactionIsolationLevel -> System.Data.IsolationLevel
+        // From https://github.com/FrendsPlatform/Frends.Sql/blob/master/Frends.Sql/Extensions.cs
+        internal static IsolationLevel GetSqlTransactionIsolationLevel(this SqlTransactionIsolationLevel sqlTransactionIsolationLevel)
+        {
+            return GetEnum<IsolationLevel>(sqlTransactionIsolationLevel);
+        }
+        // Return value or 0, depending on whether this bool is true or false, respectively.
+        // From https://github.com/FrendsPlatform/Frends.Sql/blob/master/Frends.Sql/Extensions.cs
+        public static T GetFlag<T>(this bool value, T flag)
+        {
+            return value ? flag : default(T);
+        }
+        // Get inserted row count with reflection
+        // http://stackoverflow.com/a/12271001
+        internal static int RowsCopiedCount(this SqlBulkCopy bulkCopy)
+        {
+            const string rowsCopiedFieldName = "_rowsCopied";
+            FieldInfo rowsCopiedField = null;

```

Comment:
You don't have to define this variable here, but you can define it at the same time as you set a value to the variable. So this line can be removed and `FieldInfo` can be added to the beginning of the below line.